include/zephyr/senss/senss.h

```diff
@@ -0,0 +1,522 @@
+ * model, if a virtual sensor's required dependency sensor type not found, will
+ * return error.
+ * Then call each sensor instances' init callback API.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+int senss_init(void);
+
+ /**
+  * @brief De-initialize sensor subsystem core.
+  *
+  * This will release all dynamically allocated internal resources, and call
+  * all sensor instances' deinit callback API.
+  *
+  * @return 0 on success or negative error value on failure.
+  */
+int senss_deinit(void);
+
+ /**
+  * @brief Find sensor instances by sensor type and instance ID.
+  *
+  * Application can use these API to find sensor instances information by
+  * sensor types and instance id, or enumerate all sensor instances'
+  * information at one call.
+  *
+  * This API just returned read only information of sensor instances,
+  * no side effect to sensor instances.
+  *
+  * @param type The sensor type which need to find,
+  * \ref SENSS_SENSOR_TYPE_ALL for all types.
+  *
+  * @param instance The instance ID which need to find, -1 for find all
+  * instances with type \p type
+  *
+  * @param max The max count of the \p infos array input. Can get real count
+  * number via \ref senss_get_sensor_count.
+  *
+  * @param *info Input info array for receiving found sensor instances
+  *
+  * @return Number of sensor instances found, 0 returned if no found
+  */
+int senss_find_sensor(int type, int instance,
+		 int max, struct senss_sensor_info *info);
+
+ /**
+  * @brief Get sensor instance count of given type and instance ID.
+  *
+  * @param type The sensor type for checking, \ref SENSS_SENSOR_TYPE_ALL
+  * for all types.
+  *
+  * @param instance The instance ID for checking, -1 for all instances with
+  * \p type type
+  *
+  * @return Count of matched sensor instances, 0 returned if no matched.
+  */
+int senss_get_sensor_count(int type, int instance);
+
+/**
+ * @brief Open sensor instance by sensor type and instance ID.
+ *
+ * Application clients use it to open a sensor instance and get its handle.
+ * Support multiple Application clients for open same sensor instance,
+ * in this case, the returned handle will different for different clients.
+ *
+ * @param type The sensor type which need to find.
+ *
+ * @param instance The instance ID which need to open.
+ *
+ * @param *handle The opened instance handle, if failed will be set to NULL.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_open_sensor(int type, int instance,
+		senss_sensor_handle_t *handle);
+
+/**
+ * @brief Close sensor instance.
+ *
+ * @param handle The sensor instance handle need to close.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_close_sensor(senss_sensor_handle_t handle);
+
+
+/**
+ * @brief Set custom configuration from a sensor instance.
+ *
+ * Some sensor need support customized configuration, calibration data etc.
```

Comment:
Could the configuration be documented in the device tree and applied to the sensor using the existing API (resolution, set_bias) functions.